File: algorithm/practice/sorting&search/(E)1_twoSum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> myMap = new HashMap<>();
        int expect = 0;

        for(int i = 0; i <= nums.length - 1; i++){
            expect = target - nums[i];

            if(myMap.containsKey(expect)){
                return new int[] {myMap.get(expect), i};
            }

            myMap.put(nums[i], i);
        }

        return new int[] {};
    }
}

File: algorithm/practice/sorting&search/(E)1_twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let myMap = new Map();
    let expect = 0;

    for(let i = 0; i <= nums.length - 1; i++){
        expect = target - nums[i];

        if(myMap.has(expect)){
            return  [myMap.get(expect), i];
        }
        myMap.set(nums[i], i);
    }
};